docs/dependency-management.md

```diff
@@ -0,0 +1,50 @@
+# Dependency Management Guidelines
+
+This repo uses a number of third-party rust crates, so we need to be mindful
+of how these dependencies will affect our consumers.  Considerations include:
+
+* General code quality
+* [Licensing compatibility](https://www.mozilla.org/en-US/MPL/license-policy/#Licenses_Compatible_with_the_MPL)
+* Handling of security vulnerabilities
+* The potential for [supply-chain compromise](https://medium.com/intrinsic/compromised-npm-package-event-stream-d47d08605502)
+
+We're still evolving our policies in this area, but here are the
+guidelines we've developed so far:
+
+* When adding a new dependency:
+  * In general, be conservative in adding new third-party dependencies.
+    * For trivial functionality, consider just writing it yourself.
+      Remember the cautionary tale of [left-pad](https://www.theregister.co.uk/2016/03/23/npm_left_pad_chaos/).
+    * Check if we already have a crate in our dependency tree that can provide the needed functionality.
+    * Prefer widely-used crates or crates from rust-lang-nursery.
```

Comment:
Crates used in mozilla-central (with the exception of things only used as test or build dependencies), rustc, servo (except networking and crypto), etc. Are all presumably fine as well. It might not be worth calling this out explicitly though.

Reply:
Ah yes, thanks for the reminder, we had previously discussed calling this out explicitly.